drivers/modem/Kconfig.st87mxx

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2025 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+# ST87MXX driver options
+
+config MODEM_ST87MXX
+	bool "ST87MXX modem driver"
+	select MODEM_RECEIVER
+	select MODEM_CONTEXT
+	select MODEM_CMD_HANDLER
+	select MODEM_IFACE_UART
+	select MODEM_SOCKET
+	select NET_OFFLOAD
+	select NET_SOCKETS_OFFLOAD
+	imply GPIO
+	help
+	  Choose this setting to enable ST87M01 NB-IoT modem driver.
+
+if MODEM_ST87MXX
+	config MODEM_ST87MXX_INIT_PRIORITY
+	int "ST87M01 driver init priority"
+	default 80
+	help
+	  ST87M01 device driver initialization priority.
+	  Do not mess with it unless you know what you are doing.
+	  Note that the priority needs to be lower than the net stack
+	  so that it can start before the networking sub-system.
+
+	config MODEM_ST87MXX_MAX_RX_DATA_LENGTH
```

Comment:
file is not formatted correctly https://docs.zephyrproject.org/latest/contribute/style/kconfig.html